### PR TITLE
8344217: Remove calls to SecurityManager and doPrivileged in java.net.DatagramSocket and java.net.NetMulticastSocket after JEP 486 integration

### PR DESCRIPTION
**Pull request by dfuch**
Please find here a patch that removes use of SecurityManager and doPrivileged in DatagramSocket/MulticastSocket implementation.

Some allusion to the SecurityManager was missed in DatagramSocket::connect, so this patch contains a small API documentation change that will require a CSR.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8344777](https://bugs.openjdk.org/browse/JDK-8344777) to be approved

### Issues
 * [JDK-8344217](https://bugs.openjdk.org/browse/JDK-8344217): Remove calls to SecurityManager and doPrivileged in java.net.DatagramSocket and java.net.NetMulticastSocket after JEP 486 integration (**Sub-task** - P4)
 * [JDK-8344777](https://bugs.openjdk.org/browse/JDK-8344777): Remove calls to SecurityManager and doPrivileged in java.net.DatagramSocket and java.net.NetMulticastSocket after JEP 486 integration (**CSR**)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**) Review applies to [668e0f99](https://git.openjdk.org/jdk/pull/22303/files/668e0f9915c50218b8b7e3f7db469b0414d00cea)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) Review applies to [668e0f99](https://git.openjdk.org/jdk/pull/22303/files/668e0f9915c50218b8b7e3f7db469b0414d00cea)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22303/head:pull/22303` \
`$ git checkout pull/22303`

Update a local copy of the PR: \
`$ git checkout pull/22303` \
`$ git pull https://git.openjdk.org/jdk.git pull/22303/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22303`

View PR using the GUI difftool: \
`$ git pr show -t 22303`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22303.diff">https://git.openjdk.org/jdk/pull/22303.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22303#issuecomment-2492015992)
</details>

### CODE REVIEW

**Review by RogerRiggs — APPROVED** [review]
looks fine.

**Review by jaikiran — APPROVED** [review]

File: src/java.base/share/classes/java/net/DatagramSocket.java [hunk 1]
```diff
@@ -387,12 +387,8 @@ public void bind(SocketAddress addr) throws SocketException {
...
      * that it was previously connected to.
      *
-     * <p> Care should be taken to ensure that a connected datagram socket
-     * is not shared with untrusted code. When a socket is connected,
-     * {@link #receive receive} and {@link #send send} <b>will not perform
-     * any security checks</b> on incoming and outgoing packets, other than
-     * matching the packet's and the socket's address and port. On a send
-     * operation, if the packet's address is set and the packet's address
+     * <p> When a socket is connected, additional checks are performed
+     * on a send operation: if the packet's address is set and the packet's address
```
[the remark this answers is not in the document]
Comment (dfuch) [reply src/java.base/share/classes/java/net/DatagramSocket.java]: Thanks Alan. Done. I have also updated the CSR.

File: src/java.base/share/classes/java/net/NetMulticastSocket.java [hunk 2]
```diff
@@ -333,41 +301,13 @@ public synchronized void receive(DatagramPacket p) throws IOException {
...
-                        }
-                    } // end of while
-                }
-            }
             DatagramPacket tmp = null;
+            // explicitFilter may be set to 'true' at connect() time and will
+            // be set to 'false' in disconnect() - or when there's no more
+            // pending packets to filter. If explicitFilter is true,
+            // it means we're connected.
             if (explicitFilter) {
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/net/NetMulticastSocket.java:309]: Maybe some day we will be able to re-visit this and just have connect flush any pending datagrams.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**